app/scripts/app.js

```diff
@@ -90,13 +90,21 @@ angular.module('yourStlCourts').config(function ($stateProvider, $urlRouterProvi
         citations: {value: undefined}
       },
       resolve: {
-        citations: function ($stateParams,Session) {
-          var citations = Session.getLatestCitations();
-          if ($stateParams.citations){
-            citations = $stateParams.citations;
-            Session.setLatestCitations(citations);
-          }
-          return citations;
+        citations: function ($stateParams,Session,Courts) {
```

Comment:
This really isn't the place for this. Don't add `courtName` to the citation itself but search for it in a similar loop when you want to display it in the list or in the citation page with a controller function. If to do that you need to resolve `courts` via `Courts.findAll()`, that's fine. but there is too much logic going on here in this step.